syft/pkg/cataloger/javascript/parse_node_binary.go

```diff
@@ -0,0 +1,43 @@
+package javascript
+
+import (
+	"regexp"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/source"
+)
+
+var nodeClassifier = generic.Classifier{
+	Package: "node.js", // Note: this purposely matches the "node.js" string to aid nvd vuln matching
+	FilepathPatterns: []*regexp.Regexp{
+		// note: should we just parse all files resolved with executable mimetypes
+		// regexp that matches node binary
+		regexp.MustCompile(`(.*/|^)node$`),
+	},
+	EvidencePatternTemplates: []string{
+		// regex that matches node.js/vx.y.z
+		`(?m)node\.js\/v(?P<version>[0-9]+\.[0-9]+\.[0-9]+)`,
```

Comment:
Could this just be a regex to keep this change more targeted to what it's doing? I don't _see_ any templating that's needed here and it would simplify this bit here: https://github.com/anchore/syft/pull/1296/files#diff-4ed9ad02a509d1b6975f51c8902c2bda275ee91011f73ef48bca5b3249358724R53-R72